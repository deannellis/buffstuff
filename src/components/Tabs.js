import React from 'react';

const Tabs = ({ labels, activeIndex, handleSelect, children }) => {
    return (
        <>
            <div className="tabs__tablist">
            {
                labels.map((label, i) => (
                    <a className={activeIndex == i ? 'tabs__tab--active tabs__tab' : 'tabs__tab'} onClick={() => {handleSelect(i)}} key={`tab-${i}`} >{label}</a>
                ))
            }
            </div>
            <div className="tabs__tab-content">
                {children}
            </div>
        </>
    );
}
Tabs.defaultProps = {
    labels: ['default prop'],
    activeIndex: 0,
    handleSelect: () => {console.log('handle select (default prop)')}
}
 
export default Tabs;