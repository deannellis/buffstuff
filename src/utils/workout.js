import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
