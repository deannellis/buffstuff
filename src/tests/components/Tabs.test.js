import React from 'react';
import { shallow } from 'enzyme';

import Tabs from '../../components/Tabs';
import { weights } from '../fixtures/workout';

const defaultHandleSelect = () => {};

test('should render Tabs with required props', () => {
	const labels = Object.keys(weights);
	delete labels.chinup;
	const wrapper = shallow(
		<Tabs labels={labels} handleSelect={defaultHandleSelect} />
	);
	expect(wrapper).toMatchSnapshot();
});

test('should render Tabs with props', () => {
	const labels = Object.keys(weights);
	delete labels.chinup;
	const wrapper = shallow(
		<Tabs labels={labels} activeIndex={1} handleSelect={defaultHandleSelect} />
	);
	expect(wrapper).toMatchSnapshot();
});
