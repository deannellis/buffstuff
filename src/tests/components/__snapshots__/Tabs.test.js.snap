// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Tabs with props 1`] = `
<Fragment>
  <div
    className="tabs__tablist"
  >
    <a
      className="tabs__tab"
      key="tab-0"
      onClick={[Function]}
    >
      bench
    </a>
    <a
      className="tabs__tab--active tabs__tab"
      key="tab-1"
      onClick={[Function]}
    >
      row
    </a>
    <a
      className="tabs__tab"
      key="tab-2"
      onClick={[Function]}
    >
      squat
    </a>
    <a
      className="tabs__tab"
      key="tab-3"
      onClick={[Function]}
    >
      deadlift
    </a>
    <a
      className="tabs__tab"
      key="tab-4"
      onClick={[Function]}
    >
      overhead
    </a>
    <a
      className="tabs__tab"
      key="tab-5"
      onClick={[Function]}
    >
      chinup
    </a>
  </div>
  <div
    className="tabs__tab-content"
  />
</Fragment>
`;

exports[`should render Tabs without props 1`] = `
<Fragment>
  <div
    className="tabs__tablist"
  >
    <a
      className="tabs__tab--active tabs__tab"
      key="tab-0"
      onClick={[Function]}
    >
      default prop
    </a>
  </div>
  <div
    className="tabs__tab-content"
  />
</Fragment>
`;
